### PR TITLE
refactored handle_exceptions decorator

### PR DESCRIPTION
**Pull request by Madjid-CH**
# PR Description

I saw that the handle_exceptions decorator is starting to get very long so I took the liberty in adding some tests to it and refactoring it. 
Of course if this seems unnecessary we can just forget about this idea😄

### CODE REVIEW

File: ivy/utils/exceptions.py [hunk 1]
```diff
@@ -349,58 +357,20 @@ def _handle_exceptions(*args, **kwargs):
...
-            raise ivy.utils.exceptions.IvyValueError(
-                fn.__name__, str(e), include_backend=True
-            )
-        except IvyDeviceError as e:
-            _configure_stack_trace(e.__traceback__)
-            raise ivy.utils.exceptions.IvyDeviceError(
-                fn.__name__, str(e), include_backend=True
-            )
-        except InvalidBackendException as e:
+        except (IvyNotImplementedException, InvalidBackendException) as e:
```
Comment (vedpatwardhan) [inline ivy/utils/exceptions.py:360]: Not sure why we have a separate except block for these 2 exceptions. I think we can just add `IvyNotImplementedException` to the `_non_ivy_exception_mapping` and handle `InvalidBackendException` through the `IvyException` except block

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
Hey @Madjid-CH, this will be a great addition to the codebase actually, just some minor changes needed, thanks 😄

File: ivy/utils/exceptions.py (hunk 1 above)
Comment (Madjid-CH) [reply ivy/utils/exceptions.py:360]: okay, I'll do that

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge, thanks @Madjid-CH 😄
(There's a few tests failing in the CI, but they seem unrelated from what I see)

**Pull request merged**